### ISSUE TITLE
Closely watch google-analytics tracking

### ISSUE THREAD

**Issue by jenlampton** [backdrop-ops/backdropcms.org#127]
What should we change about the current tracking? Do we need to set up specific events?

Who is going to manage our GA results and report feedback?
* volunteers?

Follow-up from https://github.com/backdrop-ops/backdropcms.org/issues/122

**Comment by sutibun** [comment]
\*raises hand\* Monkey can look over GA.

Though are you sure GA is installed and working on https://backdropcms.org/ ? I went to the homepage, looked at the source code, and don't see the usual GA tracking code.

Don't think we need to set up specific events right now (wouldn't know how anyway)

**Comment by thejimbirch** [comment]
I can help design and implement Events and Goals in GA.

**Comment by sutibun** [comment]
> I can help design and implement Events and Goals in GA

Sweet. Out of curiosity @thejimbirch Do you have any particular goals & events in mind to create at the top of your head?

**Comment by thejimbirch** [comment]
## Goals
Goals would be used to track the primary goals of the website.
- Download Backdrop Click
- Set up on Pantheon Click
- "Show Support" Submit
- "Keep me Informed" Submit
- "Contact Form" Submit

## Events
Events are usually used to track the actions of users on the site.  
- Main Menu Click
- - "About Backdrop" Click
- - "User Guide" Click
- - "Developer Guide" Click
- "We support" Click

**Comment by sutibun** [comment]
@thejimbirch Nice! Didn't think of all those. Tracking "Download backdrop" escaped me. Something so obvious. 

> Events are usually used to track the actions of users on the site.

And you helped me understand Events!

You've def also helped clarify my thinking. I think the following metrics will be most important to keep an eye on for Bd

- Bd Downloads
- Donation &
- Email newsletters sign ups

**Comment by quicksketch** [comment]
> Though are you sure GA is installed and working on https://backdropcms.org/ ? I went to the homepage, looked at the source code, and don't see the usual GA tracking code.

I think you may be right @sutibun. I couldn't find any code on the site either. We'll set this up and get back to this issue. Anyone want to volunteer to install and set up GA on the code-side?

**Comment by sutibun** [comment]
> Anyone want to volunteer to install and set up GA on the code-side?

If by that you mean working with Drupal templates... prob not the monkey. Can try but prob gonna take a long time and might mess something up. ^^;

But if you mean [via a module](https://backdropcms.org/project/googleanalytics) then ...

\*raises hand\*

**Comment by thejimbirch** [comment]
Google Analytics have been added to both backdropcms.org and api.backdropcms.org

**Comment by quicksketch** [comment]
Yay! Thanks @thejimbirch for pushing us forward on this. Are you familiar with setting up goals in GA and adding GA event triggers to the HTML? Maybe we should get you set up with a Git checkout of the BackdropCMS.org site and database dump.

**Comment by thejimbirch** [comment]
Yes.  I could use a DB dump.  I got the main site set up locally last night and started to work on the event triggers in JS.  Will hopefully have time to complete tonight.

**Comment by thejimbirch** [comment]
Re: Goals.  Upon further inspection, I don't see much opportunity for Goals.  Goals need landing pages, for which most of the actual primary goals of the site do not have (Zip Download), or are offsite  Pantheon, Mailchimp).

I am working on implementing these Events:

Category > Action > Label
====================
Backdrop > Download > Page event happened on (document.title)
Backdrop > Pantheon > document.title
Newsletter > Subscribe > document.title
Support > Submit > document.title
Contact > Submit > document.title

**Comment by thejimbirch** [comment]
I updated my Pull Request.  I realized the original one I created had extraneous files in there.

@quicksketch Please let me know when you are able to pull and I will test the GA connection.

**Comment by jenlampton** [comment]
@thejimbirch I had to remove the js file since it was preventing user registration. Let's do some more testing (and maybe add back those commented out functions?) and once we have it all working smoothly we can add it back in.

**Comment by jenlampton** [comment]
I'd love to revisit the javascript that tracks searches etc.  @thejimbirch let me know if you want to spend some time together troubleshooting the code we have to get it all working smoothly :)

**Comment by thejimbirch** [comment]
Sorry for the delay Jen, I have been swamped with work.

Probably need a different approach for the on-site events.  The script I was using is good for off-site and file links as it takes the click, registers it and then sends the user on their way.   I can see how that would mess up the user reg.

**Comment by jenlampton** [comment]
No worries about the delay, we appreciate your help whenever you have time to give it!! :)  Let's brainstorm an approach that will work - maybe another Drupal site has a good example?

**Comment by Dinornis** [comment]
In the spirit of supporting Open Source software, have you considered using Piwik?

What is Piwik?
Piwik is a free (GPL licensed) web analytics platform, an open source alternative to Google Analytics.
[Drupal 6, 7 and 8 stable versions](https://www.drupal.org/project/piwik) of Piwik Web Analytics are available.
